### PR TITLE
Implement CMAD feature

### PR DESCRIPTION
**Pull request by moataz-mhmd**
*Description of changes:*
# Added CMAD features Changes

The release of Customer Managed Active Directory introduces four new parameters available for several CLI commands that create or modify instances. All four parameters are required to be specified in order to join the domain:

|Parameter	|Type	|Description	|Constraints	|Example	|
|---	|---	|---	|---	|---	|
|domain-fqdn	|String	|The fully qualified domain name of your active directory.	|Cannot be greater than 64 characters	|domain.corp	|
|domain-ou	|String	|The organizational unit for your DB instance to join in distinguished name format. 	|Cannot be greater than 64 characters	|OU=awsComputers,DC=domain,DC=corp	|
|domain-auth-secret-arn	|String	|The ARN for the AWS Secrets Manager secret that contains the credentials for the user performing the domain join.	|-	|- 	|
|domain-dns-ips	|List (String)	|The DNS IPs in IPV4 format for your active directory domain controllers. In case you only have one IP, use the same IP for both entries in the list.	|2 IPs must be provided	|“1.1.1.1" "2.2.2.2"	|

The updated CLI commands include:

* [create-db-instance](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance.html)
* [create-db-instance-read-replica](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance-read-replica.html)
* [restore-db-instance-from-db-snapshot](https://docs.aws.amazon.com/cli/latest/reference/rds/restore-db-instance-from-db-snapshot.html)
* [restore-db-instance-to-point-in-time](https://docs.aws.amazon.com/cli/latest/reference/rds/restore-db-instance-to-point-in-time.html)
* [modify-db-instance](https://docs.aws.amazon.com/cli/latest/reference/rds/modify-db-instance.html)

To remove an instance from a Customer Managed Active Directory, the existing method of specifying the `--domain` parameter as `none` in a `modify-db-instance` CLI command continues to work. 

In addition, a new flag is introduced to the `modify-db-instance` CLI command that removes the domain from the instance:

|Parameter	|Type	|Description	|Constraints	|Example	|
|---	|---	|---	|---	|---	|
|disable-domain	|Boolean	|If present, removes the instance from the domain	|-	|-	|

It is necessary to submit all four parameters, regardless of whether only one of them has changed. Therefore, we will not utilize the "diff" function in the modify call.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-rds-dbinstance/aws-rds-dbinstance.json [hunk 1]
```diff
@@ -232,10 +232,29 @@
...
+    "DomainDnsIps": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      },
+      "description": "The IPv4 DNS IP addresses of your primary and secondary Active Directory domain controllers."
+    },
+    "DomainFqdn": {
+      "type": "string",
+      "description": "Specifies the fully qualified domain name of an Active Directory domain."
```
Comment (dbbh) [inline aws-rds-dbinstance/aws-rds-dbinstance.json:248]: Public docs are slightly different
`The fully qualified domain name (FQDN) of an Active Directory domain.`

File: aws-rds-dbinstance/aws-rds-dbinstance.json [hunk 2]
```diff
@@ -232,10 +232,29 @@
       "type": "string",
       "description": "The Active Directory directory ID to create the DB instance in. Currently, only MySQL, Microsoft SQL Server, Oracle, and PostgreSQL DB instances can be created in an Active Directory Domain."
     },
+    "DomainAuthSecretArn": {
+      "type": "string",
+      "description": "The ARN for the Secrets Manager secret that contains the credentials for the user performing the domain join."
```
Comment (dbbh) [inline aws-rds-dbinstance/aws-rds-dbinstance.json:237]: Here as well
`The ARN for the Secrets Manager secret with the credentials for the user joining the domain.`

File: aws-rds-dbinstance/aws-rds-dbinstance.json [hunk 3]
```diff
@@ -232,10 +232,29 @@
...
+      },
+      "description": "The IPv4 DNS IP addresses of your primary and secondary Active Directory domain controllers."
+    },
+    "DomainFqdn": {
+      "type": "string",
+      "description": "Specifies the fully qualified domain name of an Active Directory domain."
+    },
     "DomainIAMRoleName": {
       "type": "string",
       "description": "Specify the name of the IAM role to be used when making API calls to the Directory Service."
```
Comment (dbbh) [inline aws-rds-dbinstance/aws-rds-dbinstance.json:252]: Seems this one as well - https://awscli.amazonaws.com/v2/documentation/api/latest/reference/rds/modify-db-instance.html
`
    The name of the IAM role to use when making API calls to the Directory Service.

    This setting doesn’t apply to RDS Custom DB instances.
`

**Review by dbbh — APPROVED** [review]

File: aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java [hunk 4]
```diff
@@ -663,9 +695,18 @@ public static ResourceModel.ResourceModelBuilder translateDbInstanceFromSdkBuild
...
         String domain = null;
         String domainIAMRoleName = null;
+        String domainAuthSecretArn = null;
+        String domainFqdn = null;
+        String domainOu = null;
+        List<String> domainDnsIps = null;
         if (CollectionUtils.isNotEmpty(dbInstance.domainMemberships())) {
-            domain = dbInstance.domainMemberships().get(0).domain();
-            domainIAMRoleName = dbInstance.domainMemberships().get(0).iamRoleName();
+            DomainMembership domainMembership = dbInstance.domainMemberships().get(0);
```
Comment (khebul) [inline aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java:703]: If this is not a CMAD, but AWS managed domain - will those properties be properly populated?

**Review by dbbh — APPROVED** [review]

**Review by dbbh — APPROVED** [review]

**Review by khebul — APPROVED** [review]

**Pull request merged**